server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -682,13 +722,6 @@ public void batchDelete(List<Pool> pools) {
         for (Pool pool : pools) {
             currentSession().delete(pool);
         }
-        for (Pool pool : pools) {
-            // Maintain runtime consistency. The entitlements for the pool
-            // have been deleted on the database because delete is cascaded on
-            // Pool.entitlements relation
-            Hibernate.initialize(pool.getEntitlements());
```

Comment:
Could we log a warning that this method should not be used with initialized entitlement collections? If we don't log a warning a developer will never know they did something wrong.